core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceSpec.scala

```diff
@@ -404,135 +393,186 @@ class MultiCloudWorkspaceServiceSpec extends AnyFlatSpec with Matchers with Opti
-        mcWorkspaceService.cloneMultiCloudWorkspace(
-          mock[WorkspaceService](RETURNS_SMART_NULLS),
-          testData.azureWorkspace.toWorkspaceName,
-          WorkspaceRequest(
-            "fake_mc_billing_project_name",
-            UUID.randomUUID().toString,
-            Map.empty,
-            None,
-            None,
-            None,
-            None
-          )
-        ),
-        Duration.Inf
-      )
-    }
-
-    result.errorReport.statusCode.value shouldBe StatusCodes.NotImplemented
-  }
-
-  it should "not allow users to clone azure workspaces into gcp billing projects" in {
-    val workspaceManagerDAO = new MockWorkspaceManagerDAO()
-    val config = MultiCloudWorkspaceConfig(ConfigFactory.load())
-    val bpmDAO = mock[BillingProfileManagerDAO](RETURNS_SMART_NULLS)
-
-    val mcWorkspaceService = spy(
-      MultiCloudWorkspaceService.constructor(
-        slickDataSource,
-        workspaceManagerDAO,
-        bpmDAO,
-        new MockSamDAO(slickDataSource),
-        config,
-        workbenchMetricBaseName
-      )(testContext)
-    )
-
-    doReturn(Future.successful(testData.billingProject))
+    doReturn(Future.successful(testData.workspace))
       .when(mcWorkspaceService)
-      .getBillingProjectContext(any(), any())
+      .getWorkspaceContext(equalTo(testData.workspace.toWorkspaceName), any())
 
-    doReturn(Future.successful(testData.azureWorkspace))
-      .when(mcWorkspaceService)
-      .getWorkspaceContext(equalTo(testData.azureWorkspace.toWorkspaceName), any())
+    runTest(mcWorkspaceService)
+  }
 
-    val result = intercept[RawlsExceptionWithErrorReport] {
-      Await.result(
-        mcWorkspaceService.cloneMultiCloudWorkspace(
-          mock[WorkspaceService](RETURNS_SMART_NULLS),
-          testData.azureWorkspace.toWorkspaceName,
-          WorkspaceRequest(
-            testData.billingProject.projectName.value,
-            UUID.randomUUID().toString,
-            Map.empty,
-            None,
-            None,
-            None,
-            None
+  it should "fail if the destination workspace already exists" in
+    withEmptyTestDatabase {
+      withMockedMultiCloudWorkspaceService { mcWorkspaceService =>
+        val result = intercept[RawlsExceptionWithErrorReport] {
+          Await.result(
+            for {
+              _ <- slickDataSource.inTransaction { access =>
+                access.rawlsBillingProjectQuery.create(testData.billingProject) >>
+                  access.workspaceQuery.createOrUpdate(testData.azureWorkspace)
+              }
+              _ <- mcWorkspaceService.cloneMultiCloudWorkspace(
+                mock[WorkspaceService](RETURNS_SMART_NULLS),
+                testData.azureWorkspace.toWorkspaceName,
+                WorkspaceRequest(
+                  testData.azureWorkspace.namespace,
+                  testData.azureWorkspace.name,
+                  Map.empty
+                )
+              )
+            } yield fail(),
```

Comment:
Why `yield fail()`?

Reply:
Because if it gets to this point the test has failed